app/services/messaging/twilio/api.rb

```diff
@@ -0,0 +1,57 @@
+class Messaging::Twilio::Api < Messaging::Channel
```

Comment:
Hmmmmm we've already gone two levels deep with the inheritance on here but there is indeed a LOT of Twilio logic. Did we explore any module-based approaches for mixing in the Twilio-specific logic? Did it look too weird?

Reply:
We did try a bunch of approaches, however nothing else represents the situation at hand better:
- There are many channels (potentially across different providers). `Channel` is the common interface.
- The Twilio API has some standard behaviour that is shared by all its channels and can be independently tested.
- `OtpSMS`, `Whatsapp` etc are Twilio channels.

This probably also works well because each layer is pretty thin. For ex. the bottom most layers are mostly doing just configuration.

Reply:
Neat neat, yeah just wanted to make sure we did our due diligence and landed up here. Very cool.